python/langsmith/_internal/_operations.py

```diff
@@ -271,3 +273,32 @@ def serialized_run_operation_to_multipart_parts_and_context(
         acc_parts,
         f"trace={op.trace_id},id={op.id}",
     )
+
+
+def compress_multipart_parts_and_context(
+    parts_and_context: MultipartPartsAndContext,
+    compressor_writer: zstd.ZstdCompressionWriter,
+    boundary: str,
+) -> None:
+    for part_name, (filename, data, content_type, headers) in parts_and_context.parts:
+        part_header = f"--{boundary}\r\n"
+        part_header += f'Content-Disposition: form-data; name="{part_name}"'
```

Comment:
We should collect these formatted parts in a list and do a single "".join() rather than copying the header string so many times

Reply:
fixed